components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthCredentialValidator.java

```
@@ -0,0 +1,478 @@
+     * @throws APISecurityException If an authentication failure or some other error occurs
+     */
+    BasicAuthCredentialValidator(SynapseEnvironment env) throws APISecurityException {
+        this.gatewayKeyCacheEnabled = isGatewayTokenCacheEnabled();
+        this.getGatewayUsernameCache();
+
+        ConfigurationContext configurationContext = ServiceReferenceHolder.getInstance().getAxis2ConfigurationContext();
+        APIManagerConfiguration config = ServiceReferenceHolder.getInstance().getAPIManagerConfiguration();
+        String url = config.getFirstProperty(APIConstants.API_KEY_VALIDATOR_URL);
+        if (url == null) {
+            throw new APISecurityException(APISecurityConstants.API_AUTH_GENERAL_ERROR,
+                    "API key manager URL unspecified");
+        }
+
+        try {
+            authAdminStub = new AuthenticationAdminStub(configurationContext, url +
+                    "AuthenticationAdmin");
+        } catch (AxisFault axisFault) {
+            throw new APISecurityException(APISecurityConstants.API_AUTH_GENERAL_ERROR, axisFault.getMessage());
+        }
+
+        try {
+            remoteUserStoreManagerServiceStub = new RemoteUserStoreManagerServiceStub(configurationContext, url +
+                    "RemoteUserStoreManagerService");
+        } catch (AxisFault axisFault) {
+            throw new APISecurityException(APISecurityConstants.API_AUTH_GENERAL_ERROR, axisFault.getMessage());
+        }
+        ServiceClient svcClient = remoteUserStoreManagerServiceStub._getServiceClient();
+        CarbonUtils.setBasicAccessSecurityHeaders(config.getFirstProperty(APIConstants.AUTH_MANAGER_USERNAME),
+                config.getFirstProperty(APIConstants.AUTH_MANAGER_PASSWORD), svcClient);
+
+        try {
+            host = new URL(url).getHost();
+        } catch (MalformedURLException e) {
+            throw new APISecurityException(APISecurityConstants.API_AUTH_GENERAL_ERROR, e.getMessage());
```

Exception has been swallowed leading to the original stack trace being lost. Pass entire exception object into wrapped exception

`catch (CaughtExecption e) {
      throw new ThrownException("Error detected", e);
}`

---

Updated